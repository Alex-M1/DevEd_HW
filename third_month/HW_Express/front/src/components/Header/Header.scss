.header-bg{
  background: #1976d2;
  padding: 10px 0;
}

.app-header{
  display: grid;
  grid-template-columns: 40% 60%;
  align-items: center;
}

.header-navigation{
  &__links{
    display: flex;
    padding: 0;
    &--item{
      list-style-type: none;
      margin-right: 100px;
      a{
        font-size: 21px;
        color: #c2c2c2;
        text-decoration: none;
      }
    }
    &--active{
      color: #f5f5f5 !important;
      border-bottom: 3px solid #d21919;
    }
  }
}
