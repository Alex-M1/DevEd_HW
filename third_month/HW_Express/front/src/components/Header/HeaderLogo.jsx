import React from 'react';

function HeaderLogo(props) {
  return (
    <div className="logo">
      <img src="" alt="logo" />
      <div>BookShop</div>
    </div>
  );
}

export default HeaderLogo;