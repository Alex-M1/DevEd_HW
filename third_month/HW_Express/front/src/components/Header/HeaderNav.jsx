import React from 'react';
import { NavLink } from 'react-router-dom';

function HeaderNav(props) {
  return (
    <nav className="header-navigation">
      <ul className="header-navigation__links">
        <li className="header-navigation__links--item">
          <NavLink exact activeClassName="header-navigation__links--active" to="/">
            Books
          </NavLink>
        </li>
        <li className="header-navigation__links--item">
          <NavLink exact activeClassName="header-navigation__links--active" to="/create">
            Create Book
          </NavLink>
        </li>
      </ul>
    </nav >
  );
}

export default HeaderNav;