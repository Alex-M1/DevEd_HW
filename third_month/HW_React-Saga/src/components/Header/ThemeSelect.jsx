import React from 'react';
import { useTheme } from '../../helpers/withTheme';

function ThemeSelect() {
  const { theme, changeTheme } = useTheme();
  return (
    <select value={theme} onChange={changeTheme}>
      <option value='dark'>dark</option>
      <option value='light'>light</option>
    </select>
  );
}

export default ThemeSelect;
