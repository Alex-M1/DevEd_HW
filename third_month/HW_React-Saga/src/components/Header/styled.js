import styled from 'styled-components';

export const StHeader = styled.header`
  display: flex;
  justify-content: space-between;
  padding: 10px;
  button{
    margin-right: 10px;
  }
`;
