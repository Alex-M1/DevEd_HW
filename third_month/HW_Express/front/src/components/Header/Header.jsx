import './Header.scss';
import React from 'react';
import HeaderNav from './HeaderNav';
import HeaderLogo from './HeaderLogo';

function Header(props) {
  return (
    <header className="header-bg">
      <div className="container">
        <div className="app-header">
          <HeaderLogo />
          <HeaderNav />
        </div>
      </div>
    </header>
  );
}

export default Header;

