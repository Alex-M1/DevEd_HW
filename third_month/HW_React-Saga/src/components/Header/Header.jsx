import React from 'react';
import PropTypes from 'prop-types';
import Button from '../__common__/Button/Button';
import ThemeSelect from './ThemeSelect';
import Modal from '../Modal';
import { StHeader } from './styled';

const Header = ({ isVisible, setVisible }) => {
  console.log('123');
  const modal = isVisible ? <Modal /> : null;
  const onButtonClick = () => {
    setVisible(true);
  };
  return (
    <StHeader>
      <div>
        Person Manager
      </div>
      <div>
        <Button type="primary" onClick={onButtonClick}>
          Add person
        </Button>
        <ThemeSelect />
      </div>
      {modal}
    </StHeader>
  );
};

export default Header;

Header.propTypes = {
  isVisible: PropTypes.bool.isRequired,
  setVisible: PropTypes.func.isRequired,
};
