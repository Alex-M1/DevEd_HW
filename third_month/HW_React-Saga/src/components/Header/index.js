import { connect } from 'react-redux';
import { setVisible } from '../../store/actions';
import { modalVisible } from '../../store/selectors';
import Header from './Header';

const mapStateToProps = (state) => ({
  isVisible: modalVisible(state),
});
const mapDispatchToProps = {
  setVisible,
};

export default connect(mapStateToProps, mapDispatchToProps)(Header);
