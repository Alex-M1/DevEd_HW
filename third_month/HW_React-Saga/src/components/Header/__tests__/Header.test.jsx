import React from 'react';
import configureStore from 'redux-mock-store';
import Header from '../Header';

jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useLayoutEffect: jest.requireActual('react').useEffect,
}));
const mockStore = configureStore();
const store = mockStore({
  form: {
    name: '',
    age: '',
    phone: '',
    email: '',
    company: '',
    visible: true,
  },
  persons: [{
    name: '',
    age: '',
    phone: '',
    email: '',
    company: '',
  }],
});
const isVisible = false;
const theme = 'dark';
jest.mock('../../../helpers/withTheme', () => ({
  useTheme: jest.fn().mockReturnValue({
    theme,
    changeTheme: jest.fn().mockReturnValue(theme),
  }),
}));
describe('Header without modal', () => {
  const props = {
    isVisible,
    setVisible: jest.fn(),
  };
  it('Should match snapshot', () => {
    const component = global.shallowSmart(<Header {...props} />);
    expect(component.html()).toMatchSnapshot();
  });
  it('Should have header', () => {
    const component = global.mountSmart(<Header {...props} />);
    expect(component.find('select')).toHaveLength(1);
  });
  it('Should have Button', () => {
    const component = global.mountSmart(<Header {...props} />);
    expect(component.find('Button')).toHaveLength(1);
  });
  it('Should have div', () => {
    const component = global.mountSmart(<Header {...props} />);
    expect(component.find('div')).toHaveLength(2);
  });
  it('Should have ThemeSelect', () => {
    const component = global.mountSmart(<Header {...props} />);
    expect(component.find('ThemeSelect')).toHaveLength(1);
  });
  it('Should have click & callde setVisible', () => {
    const component = global.mountSmart(<Header {...props} />);
    component.find('button').simulate('click');
    expect(component.props().setVisible).toHaveBeenCalled();
  });
});

describe('Header with modal', () => {
  const props = {
    isVisible: !isVisible,
    setVisible: jest.fn(),
  };
  it('Should match snapshot', () => {
    const component = global.shallowSmart(<Header {...props} />, store);
    expect(component.html()).toMatchSnapshot();
  });
  it('Should have header', () => {
    const component = global.mountSmart(<Header {...props} />, store);
    expect(component.find('select')).toHaveLength(1);
  });
  it('Should have Button', () => {
    const component = global.mountSmart(<Header {...props} />, store);
    expect(component.find('Button')).toHaveLength(3);
  });
  it('Should have div', () => {
    const component = global.mountSmart(<Header {...props} />, store);
    expect(component.find('div')).toHaveLength(12);
  });
  it('Should have ThemeSelect', () => {
    const component = global.mountSmart(<Header {...props} />, store);
    expect(component.find('ThemeSelect')).toHaveLength(1);
  });
});
