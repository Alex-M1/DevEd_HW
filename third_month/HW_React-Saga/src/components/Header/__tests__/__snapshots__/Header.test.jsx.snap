// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header with modal Should match snapshot 1`] = `"<header class=\\"styled__StHeader-ycxa9o-0 fXYztV\\"><div>Person Manager</div><div><button mode=\\"primary\\" class=\\"styled__StButton-iorpwz-0 GXmVc\\">Add person</button><select><option selected=\\"\\" value=\\"dark\\">dark</option><option value=\\"light\\">light</option></select></div><div class=\\"styled__StModalBackground-k7b4xf-0 eVoaqV\\"><div class=\\"styled__StModal-k7b4xf-1 fUfvem\\"><div class=\\"styled__StModalHeader-sc-1ozqfyh-0 jmUcbk\\"><h1>Add person</h1><span>⨯</span></div><div class=\\"styled__StForm-p2kz64-0 kNCWSz\\"><div class=\\"styled__StFormItem-p2kz64-1 kSxDuO\\"><span>Имя</span><input value=\\"\\" placeholder=\\"Введите имя\\"/></div><div class=\\"styled__StFormItem-p2kz64-1 kSxDuO\\"><span>Возраст</span><input value=\\"\\" placeholder=\\"Введите возраст\\"/></div><div class=\\"styled__StFormItem-p2kz64-1 kSxDuO\\"><span>Телефон</span><input value=\\"\\" placeholder=\\"Введите телефон\\"/></div><div class=\\"styled__StFormItem-p2kz64-1 kSxDuO\\"><span>Email</span><input value=\\"\\" placeholder=\\"Введите email\\"/></div><div class=\\"styled__StFormItem-p2kz64-1 kSxDuO\\"><span>Компания</span><input value=\\"\\" placeholder=\\"Название компании\\"/></div></div><div class=\\"styled__StBtnGroup-c54mt-0 RBvzV\\"><button mode=\\"primary\\" class=\\"styled__StButton-iorpwz-0 GXmVc\\">Отправить</button><button mode=\\"cancel\\" class=\\"styled__StButton-iorpwz-0 nuamV\\">Отменить</button></div></div></div></header>"`;

exports[`Header without modal Should match snapshot 1`] = `"<header class=\\"styled__StHeader-ycxa9o-0 fXYztV\\"><div>Person Manager</div><div><button mode=\\"primary\\" class=\\"styled__StButton-iorpwz-0 GXmVc\\">Add person</button><select><option selected=\\"\\" value=\\"dark\\">dark</option><option value=\\"light\\">light</option></select></div></header>"`;
